tree/ntuple/v7/src/RColumn.cxx

```diff
@@ -110,7 +114,16 @@ void ROOT::Experimental::Internal::RColumn::MapPage(const NTupleSize_t index)
    // Set fReadPage to an empty page before populating it to prevent double destruction of the previously page in case
    // the page population fails.
    fReadPage = RPage();
-   fReadPage = fPageSource->PopulatePage(fHandleSource, index);
+
+   const auto nTeam = fTeam->size();
+   std::size_t iTeam = 1;
+   do {
+      fReadPage = fPageSource->PopulatePage(fTeam->at(fLastGoodTeamIdx)->GetHandleSource(), index);
+      if (fReadPage.IsValid())
+         break;
+      fLastGoodTeamIdx = (fLastGoodTeamIdx + iTeam++) % nTeam;
```

Comment:
Hm, does this work as expected with more than two column representations? Given `fLastGoodTeamIdx = 0` and `iTeam = 1` in the first iteration, this will progress correctly to `fLastGoodTeamIdx = 1` but then advance to `fLastGoodTeamIdx = 3`, no?

Reply:
Thanks! Good catch.